queries.md

```diff
@@ -3,41 +3,80 @@
+{ partners: { $exists: false } }
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
+{ category_code: {$eq: null}}
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
+{ $and: [ { number_of_employees :{ $lte: 1000 } }  , { number_of_employees :{ $gte: 100 } } ] }
+{name:1, number_of_employees:1,_id:0}
+
 
 ### 9. Order all the companies by their IPO price descendently.
+{'ipo.valuation_amount':-1}
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
+{number_of_employees:1,_id:0}
+{number_of_employees:-1}
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
+{ $and: [ { founded_month :{ $lte: 12 } }  , { founded_month :{ $gte: 6} } ] }
+1000
 
 ### 12. All the companies that have been 'deadpooled' after the third year.
+{ deadpooled_year: {$gte: 3}}
+{deadpooled_year:1}
 
 ### 13. All the companies founded before 2000 that have and acquisition amount of more than 10.000.000
+{ $and: [ { founded_year :{ $gte: 2000 } }  , { 'acquisition.price_amount' :{ $gte: 10000000 } } ] }
 
 ### 14. All the companies that have been acquired after 2015, order by the acquisition amount, and retrieve only their `name` and `acquisiton` field.
+{'acquisition.acquired_year':{$gte:2015}}
+{name:1,_id:0,acquisition:1}
+{acquisition:1}
 
 ### 15. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
+{name:1, founded_year:1,_id:0}
+{founded_year:1}
 
 ### 16. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `aquisition price` descendently. Limit the search to 10 documents.
+{founded_day:{$lte:7}}
+{'acquisition.price_amount':-1}
 
 ### 17. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascendant order.
+{ $and: [ { category_code :'web' }  , { number_of_employees :{ $gte: 4000} } ] }
+{number_of_employees:1}
 
 ### 18. All the companies which their acquisition amount is more than 10.000.000, and currency are 'EUR'.
+{ $and: [ { 'acquisition.price_amount' :{ $gte: 10000000} }  , { 'acquisition.price_currency_code' :'EUR' } ] }
 
 ### 19. All the companies that have been acquired on the first trimester of the year. Limit the search to 10 companies, and retrieve only their `name` and `acquisition` fields.
+{'acquisition.acquired_month':{$lte:3}}
+{name:1,acquisition:1,_id:0}
 
-### 20. All the companies that have been founded between 2000 and 2010, but have not been acquired before 2011.
\ No newline at end of file
+### 20. All the companies that have been founded between 2000 and 2010, but have not been acquired before 2011.
+{ $and: [ { founded_year :{ $gte: 2000 } }  , {founded_year :{ $lte: 2010 } },{'acquisition.acquired_year':{$ne:2011}} ] }
```

Comment:
{$and:[  {founded_year:{$gte:2000,$lte:2010}} ,{$or:[{ 'acquisition.acquired_year':{$gte:2011}  },{acquisition:{$eq:"null"} }] }] }